app/code/Magento/MediaGalleryUi/Model/SearchCriteria/CollectionProcessor/FilterProcessor/Keyword.php

```diff
@@ -77,4 +78,20 @@ private function getAssetIdsByKeyword(string $value): array
             )
         );
     }
+
+    /**
+     * Split the request string
+     *
+     * @param string $value
+     * @return array
+     */
+    private function splitRequestString(string $value): array
+    {
+        $keywordValues = [];
+        if (!is_array($value)) {
+            $keywordValues = explode(',', $value);
```

Comment:
I think the logic to expload keywords should be enhanced to support the following search input formats:

- `keyword1,keyword2`
- `keyword1 keyword2`
- `keyword1, keyword2`
- `keyword1. keyword2`
- `keyword1; keyword2`

Also, any characters not allowed for a keyword may be removed from the keywords string to provide higher chance of the match. (see `app/code/Magento/MediaGalleryUi/view/adminhtml/web/js/validation/validate-image-keyword.js`)

By the way, it would be good to ensure the punctuation marks are not allowed to be used for keywords (as we are using them to split keywords search query)

@chalov-anton what do you think? Is the list of formats complete or should we cover some other cases?

Reply:
@sivaschenko Yes, I think that is the complete list. Probably, the most commonly used variants will be

Reply:
Thanks @chalov-anton . @yolouiese let's handle only `,` and empty spaces between keywords